pkg/api/handlers/mcpwebhookvalidation.go

```diff
@@ -0,0 +1,180 @@
+package handlers
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+
+	"github.com/gptscript-ai/go-gptscript"
+	"github.com/obot-platform/obot/apiclient/types"
+	"github.com/obot-platform/obot/pkg/api"
+	v1 "github.com/obot-platform/obot/pkg/storage/apis/obot.obot.ai/v1"
+	"github.com/obot-platform/obot/pkg/system"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type MCPWebhookValidationHandler struct{}
+
+func NewMCPWebhookValidationHandler() *MCPWebhookValidationHandler {
+	return &MCPWebhookValidationHandler{}
+}
+
+func (m *MCPWebhookValidationHandler) List(req api.Context) error {
+	var list v1.MCPWebhookValidationList
+	if err := req.List(&list); err != nil {
+		return fmt.Errorf("failed to list mcp webhook validations: %w", err)
+	}
+
+	items := make([]types.MCPWebhookValidation, 0, len(list.Items))
+	for _, item := range list.Items {
+		credMap, err := req.GPTClient.RevealCredential(req.Context(), []string{system.MCPWebhookValidationCredentialContext}, item.Name)
+		if err != nil {
+			return fmt.Errorf("failed to reveal credential for webhook %s: %w", item.Name, err)
+		}
+		items = append(items, convertMCPWebhookValidation(item, credMap.Env))
+	}
+
+	return req.Write(types.MCPWebhookValidationList{Items: items})
+}
+
+func (m *MCPWebhookValidationHandler) Get(req api.Context) error {
+	var validation v1.MCPWebhookValidation
+	if err := req.Get(&validation, req.PathValue("mcp_webhook_validation_id")); err != nil {
+		return err
+	}
+
+	secretCred, err := req.GPTClient.RevealCredential(req.Context(), []string{system.MCPWebhookValidationCredentialContext}, validation.Name)
+	if err != nil {
+		return fmt.Errorf("failed to reveal credential: %w", err)
+	}
+
+	return req.Write(convertMCPWebhookValidation(validation, secretCred.Env))
+}
+
+func (m *MCPWebhookValidationHandler) Create(req api.Context) error {
+	var manifest types.MCPWebhookValidationManifest
+	if err := req.Read(&manifest); err != nil {
+		return types.NewErrBadRequest("failed to read manifest: %v", err)
+	}
+
+	if err := manifest.Validate(); err != nil {
+		return types.NewErrBadRequest("invalid manifest: %v", err)
+	}
+
+	secretCred := make(map[string]string, len(manifest.Webhooks))
+	for i, webhook := range manifest.Webhooks {
+		secretCred[strconv.Itoa(i)] = webhook.Secret
+		// Don't save the secrets in the database.
+		webhook.Secret = ""
+		manifest.Webhooks[i] = webhook
+	}
+
+	validation := v1.MCPWebhookValidation{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: system.MCPWebhookValidationPrefix,
+			Namespace:    req.Namespace(),
+		},
+		Spec: v1.MCPWebhookValidationSpec{
+			Manifest: manifest,
+		},
+	}
+
+	if err := req.Create(&validation); err != nil {
```

Comment:
This doesn't seem to return the **generated name** from system after creating to the client, if we care about that.

Reply:
It seems to return the `Name` field on the object. Just tested to make sure.